server/pxf-hive/src/main/java/org/greenplum/pxf/plugins/hive/HiveORCSearchArgumentBuilder.java

```diff
@@ -0,0 +1,328 @@
+     */
+    private boolean buildArgument(OperatorNode operatorNode) {
+
+        Operator operator = operatorNode.getOperator();
+        ColumnIndexOperand columnIndexOperand = operatorNode.getColumnIndexOperand();
+        Optional<Operand> valueOperand = operatorNode.getOperand();
+
+        ColumnDescriptor filterColumn = columnDescriptors.get(columnIndexOperand.index());
+        String filterColumnName = filterColumn.columnName();
+        Object filterValue = null;
+
+        if (valueOperand.isPresent()) {
+            // In Hive 1, boxing of values happened inside the builder
+            // For Hive 2 libraries, we need to do it before passing values to
+            // Hive jars
+
+            if (valueOperand.get() instanceof CollectionOperand) {
+                CollectionOperand collectionOperand = (CollectionOperand) valueOperand.get();
+
+                filterValue = collectionOperand
+                        .getData()
+                        .stream()
+                        .map(data -> boxLiteral(convertDataType(collectionOperand.getDataType().getTypeElem(), data)))
+                        .collect(Collectors.toList());
+            } else if (valueOperand.get() instanceof ScalarOperand) {
+                ScalarOperand scalarOperand = (ScalarOperand) valueOperand.get();
+
+                filterValue = convertDataType(scalarOperand);
+                filterValue = boxLiteral(filterValue);
+            }
+        }
+
+        PredicateLeaf.Type predicateLeafType = PredicateLeaf.Type.STRING;
+
+        if (filterValue != null) {
+            predicateLeafType = getType(filterValue);
+        }
+
+        switch (operator) {
+            case LESS_THAN:
+                filterBuilder.lessThan(filterColumnName, predicateLeafType, filterValue);
+                break;
+            case GREATER_THAN:
+                filterBuilder.startNot().lessThanEquals(filterColumnName, predicateLeafType, filterValue).end();
+                break;
+            case LESS_THAN_OR_EQUAL:
+                filterBuilder.lessThanEquals(filterColumnName, predicateLeafType, filterValue);
+                break;
+            case GREATER_THAN_OR_EQUAL:
+                filterBuilder.startNot().lessThan(filterColumnName, predicateLeafType, filterValue).end();
+                break;
+            case EQUALS:
+                filterBuilder.equals(filterColumnName, predicateLeafType, filterValue);
+                break;
+            case NOT_EQUALS:
+                filterBuilder.startNot().equals(filterColumnName, predicateLeafType, filterValue).end();
+                break;
+            case IS_NULL:
+                filterBuilder.isNull(filterColumnName, predicateLeafType);
+                break;
+            case IS_NOT_NULL:
+                filterBuilder.startNot().isNull(filterColumnName, predicateLeafType).end();
+                break;
+            case IN:
+                if (filterValue instanceof List) {
```

Comment:
```suggestion
                if (filterValue instanceof Collection) {
```

to avoid casting to List